ci/ci/pr.py

```diff
@@ -273,6 +274,7 @@ def _new_source(self, new_source):
 
     def _new_build(self, new_build):
         if self.build != new_build:
+            self.last_status = self._status_url()
```

Comment:
I think we should track the full history of target and source SHAs. Each time source or target or both change, we should note that in an array with the current build state.

We can add another page to the CI UI which is the per-PR page. This page will have a table that corresponds to this array.

Suppose a PR's history is:
1. initial push with head commit SHA `def1` while master is `abc1`
2. master moves to `abc2` (build for `abc1`-`def1` is cancelled)
3. build finishes for `abc2`-`def1` as failure
4. master moves to `abc3`
5. build finishes for `abc3`-`def1` as failure
6. feature moves to `def2`
7. build finishes for `abc3`-`def2` as success

The array will look like:
```
[ (abc1, def1, running) # added when 2 happens
, (abc2, def1, failure) # added when 4 happens
, (abc3, def1, failure) # added when 5 happens
]
```
And the page will have:

| master | feature | status |
| --- | --- | ---
| abc3 | def2 | success (link) |
| abc3 | def1 | failure (link) |
| abc2 | def1 | failure (link) |
| abc1 | def1 | incomplete |